pkg/cmd/stack/deploy.go

```diff
@@ -488,14 +488,26 @@ func deployDeployment(ctx context.Context, svcName string, s *model.Stack, c kub
 		if old.Labels[model.StackNameLabel] == "" {
 			return false, fmt.Errorf("skipping deploy of deployment '%s' due to name collision with pre-existing deployment", svcName)
 		}
-		if old.Labels[model.StackNameLabel] != s.Name {
+		if old.Labels[model.StackNameLabel] != s.Name && old.Labels[model.StackNameLabel] != "okteto" {
 			return false, fmt.Errorf("skipping deploy of deployment '%s' due to name collision with deployment in stack '%s'", svcName, old.Labels[model.StackNameLabel])
 		}
 		if v, ok := old.Labels[model.DeployedByLabel]; ok {
+			if old.Labels[model.StackNameLabel] == "okteto" {
+				d.Labels[model.DeployedByLabel] = s.Name
+			}
```

Comment:
Don't you need an else here? If you set the `deployed-by` label to the new name, the line 498 will overwrite it with the value coming from the old deployment, right?

Or moving the line 498 before doing the `okteto` check would be enough